### PR TITLE
Updated panos_registered_ip

### PR DESCRIPTION
**Pull request by mrichardson03**
#269 

Also check to see if IPs actually need to be registered/unregistered, and return changed accordingly.

### CODE REVIEW

**Comment by mrichardson03** [comment]
Question for this and #295: can you register IPs directly to a vsys?  I don't have a box to test.

**Comment by shinmog** [comment]
@mrichardson03 

Looks like when you do userid stuff, [you do](https://github.com/PaloAltoNetworks/pandevice/blob/develop/pandevice/userid.py#L126) pass in the vsys with the command.  There is a caveat with that in that you need to be setting the vsys directly on the firewall object itself instead of using a `pandevice.device.Vsys` object, but this is what the helper code is doing, so it should be fine.

So to answer your question:  yeah, include the vsys documentation fragment and do `vsys=True` with the helper constructor.  You will then have added in multi-vsys support to these two modules :)

**Comment by mrichardson03** [comment]
@shinmog 

I thought so.  I actually had it in there, just commented out.

File: library/panos_registered_ip.py [hunk 1]
```diff
@@ -58,11 +48,6 @@
...
             - List of tags that the IP address will be registered to.
         required: true
-    state:
-        description:
-            - Create or remove registered IP addresses.
-        choices: ['present', 'absent']
-        default: 'present'
 '''
 
 EXAMPLES = '''
```
Comment (shinmog) [inline library/panos_registered_ip.py]: Don't forget to update the examples.

File: library/panos_registered_ip.py [hunk 2]
```diff
@@ -121,62 +106,64 @@
...
-        password=dict(no_log=True),
-        api_key=dict(no_log=True),
-        ips=dict(type='list', required=True),
-        tags=dict(type='list', required=True),
-        state=dict(default='present', choices=['present', 'absent'])
+    helper = get_connection(
+        vsys=True,
+        with_classic_provider_spec=True,
+        with_state=True,
+        argument_spec=dict(
```
Comment (shinmog) [inline library/panos_registered_ip.py]: In the notes, you have that Panorama isn't supported.  To do this you need to specify `panorama_error` when invoking `get_connection()`.

**Pull request merged**